lib/active_model/serializable_resource.rb

```diff
@@ -41,7 +41,7 @@ def serializer
       @serializer ||=
         begin
           @serializer = serializer_opts.delete(:serializer)
-          @serializer ||= ActiveModel::Serializer.serializer_for(resource)
+          @serializer ||= ActiveModel::Serializer.serializer_for(resource, serializer_opts.slice(:_controller_class_name))
```

Comment:
I think passing the controller name to serializer_for seems weird. Would just passing the namespace make more sense? so that way you don't need to deconstantize that later?